### PR TITLE
Broadcast Messaging V2

### PR DESCRIPTION
**Pull request by abdullai-t**
#### Works with[ Admin PR](https://github.com/massenergize/frontend-admin/pull/1114)

### CODE REVIEW

File: src/api/handlers/community.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def registerRoutes(self) -> None:
     self.add("/communities.others.listForCommunityAdmin", self.list_other_communities_for_cadmin)
     self.add("/communities.listForSuperAdmin", self.super_admin_list)
     self.add("/communities.adminsOf", self.fetch_admins_of)
+    self.add("/communities.listNoPagination", self.communities_list_no_pagination)
```
Comment (BradHN1) [inline src/api/handlers/community.py:39]: I don't think you we should need routes like this. There should be a URL parameter that sets the pagination length, which could be set to return all results in one query.

File: src/api/handlers/userprofile.py [hunk 2]
```diff
@@ -36,28 +36,31 @@ def registerRoutes(self):
...
-      context: Context = request.context
-      args: dict = context.args
-      args, err = self.validator.expect("id", str, is_required=True).verify(context.args)
-      if err:
-        return err
-      visits, err = self.service.fetch_user_visits(context, args)
-      if err:
-        return err
-      return MassenergizeResponse(data=visits)
+        self.add("/users.listNoPagination", self.users_list_no_pagination)
```
Comment (BradHN1) [inline src/api/handlers/userprofile.py:49]: same comment as above; I think a URL parameter should accomplish this, not a dedicated route.

File: src/api/handlers/community.py (hunk 1 above)
Comment (abdullai-t) [reply src/api/handlers/community.py]: Thank you for the feedback

File: src/api/store/message.py [hunk 3]
```diff
@@ -29,45 +30,23 @@ def get_schedule(schedule):
...
+        com = Community.objects.filter(id=id).first()
+        if com:
+            return com.logo.file.url if com.logo else None
+        
     
 
 
-def get_message_recipients(audience, audience_type, sub_audience_type, communities):
+def get_message_recipients(audience, audience_type):
     if not is_null(audience):
```
Comment (BradHN1) [inline src/api/store/message.py:44]: if is_null returns true, do you nee to have a "return None" statment?

File: src/api/store/message.py [hunk 4]
```diff
@@ -383,16 +369,23 @@ def send_message(self, context, args) -> Tuple[dict, MassEnergizeAPIError]:
...
+            associated_community = None
             if not recipients:
                 return None, InvalidResourceError()
             
             user = get_user_from_context(context)
             if not user:
                 return None, InvalidResourceError()
             
+            if not context.user_is_super_admin:
+                associated_community = Community.objects.filter(id=communities[0]).first()
```
Comment (BradHN1) [inline src/api/store/message.py:387]: question - is there a case where multiple communities will be passed, perhaps for a community admin of multiple communities?  In that case, picking the first one won't always be correct.   (fix later)

**Review by BradHN1 — APPROVED** [review]
LGTM - test in dev

**Pull request merged**